src/commands/environments/get.ts

```diff
@@ -21,34 +22,39 @@ export default class DetailedEnvironments extends Base {
 
     public async run(): Promise<void> {
         const { flags } = await this.parse(DetailedEnvironments)
-        const keys = flags['keys']?.split(',') || []
+        const keys = flags['keys']?.split(',')
         const { headless, project } = flags
         await this.requireProject(project, headless)
 
-        let environments = await fetchEnvironments(this.authToken, this.projectKey)
-        if (keys.length) {
-            environments = environments.filter((environment) => 
-                keys.includes(environment.key) || keys.includes(environment._id))
+        if (keys) {
+            const environments = []
+            const batches = chunk(keys, 10)
+            for (const batch of batches) {
+                const batchEnvironments = await Promise.all(
+                    batch.map((key) => fetchEnvironmentByKey(this.authToken, this.projectKey, key))
+                )
+                environments.push(...batchEnvironments.filter(Boolean))
+            }
```

Comment:
not necessary, but this could be a good place to create a util function that handles batching the requests given the array and map function

Reply:
good point, i moved it into utils